src/nuclio/common/components/edit-item/edit-item.component.js

```diff
@@ -248,6 +252,24 @@
             $scope.$on('deploy-function-version', onFunctionDeploy);
         }
 
+        function updateNameValidationRules() {
```

Comment:
I'm not sure this is the way we want to go here.
I've recently worked hard on refactoring `nclEditItem` so it becomes more generic to serve all kinds of objects. It's not 100% yet, but it still went to great lengths in that direction, leaving only a few places which are custom for specific stuff (like ingresses list of HTTP trigger).

The function you wrote here “knows” about triggers, HTTP trigger, and the specific validation rule for `default-http`-named trigger.
I think it would be better to let this configuration come from the parent component (in this case `nclVersionTriggers`).
`nclEditItem` component already receives `validationRules` prop and has a `onSelectClassCallback` “event”. This is exactly two out of three things we need in order to let the parent component manage the “Name” field’s validation rules, instead of `nclEditItem` managing it internally.
Once a class is selected, `nclEditItem` “notifies” its parent via `onSelectClassCallback`, which then as a result can change the object referenced by `data-validation-rules` attribute on `<ncl-edit-item>` (in `nclVersionTriggers`’s HTML template), then `nclEditItem` needs to detect this change in `validationRules` prop and update the view accordingly.
I think the only thing missing here is to add the `$onChanges` lifecycle hook of `nclEditItem` which will detect the change in validation rules and update the view as a result.
I feel this approach is more natural, and `nclVersionTriggers` actually already declares the `ctrl.validationRules` with `itemName` in it. All it needs to do is to update it when the class is changed. But it would need to have a deep clone of `ctrl.validationRules` and make the necessary changes upon this clone, then assign this object to `ctrl.validationRules` — so `nclEditItem` could detect this change using its `$onChanges`.